### PR TITLE
[CPU][ARM] Enable f16 Interpolate on ARM

### PR DESCRIPTION
**Pull request by alvoron**
### Details:
 - Lack of f16 Interpolation leads to additional Converts in `fastseg`.

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp [hunk 1]
```diff
@@ -2113,8 +2121,13 @@ void Interpolate::initSupportedPrimitiveDescriptors() {
         pushDesc(LayoutType::nspc, undef, true, true);
         pushDesc(LayoutType::ncsp, undef, true, true);
         canUseAclExecutor = !supportedPrimitiveDescriptors.empty();
-        if (canUseAclExecutor)
+        if (canUseAclExecutor) {
             return;
+        }
+        //if ACL executor is not selected then f16 should not be used
+        if (inputPrecision == ov::element::f16) {
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:2128]: Ref Interpolate supports fp32 only. So we need to fallback on fp32 in any case, not just for fp16

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp [hunk 2]
```diff
@@ -2113,8 +2121,13 @@ void Interpolate::initSupportedPrimitiveDescriptors() {
         pushDesc(LayoutType::nspc, undef, true, true);
         pushDesc(LayoutType::ncsp, undef, true, true);
         canUseAclExecutor = !supportedPrimitiveDescriptors.empty();
-        if (canUseAclExecutor)
+        if (canUseAclExecutor) {
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:2124]: You updated only `if (is_version11) {` branch, `else` body should also be fixed.

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp [hunk 3]
```diff
@@ -2113,8 +2121,13 @@ void Interpolate::initSupportedPrimitiveDescriptors() {
...
             return;
+        }
+        //if ACL executor is not selected then f16 should not be used
+        if (inputPrecision == ov::element::f16) {
+            inputPrecision = ov::element::f32;
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:2129]: what about outputPrecision?

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp [hunk 4]
```diff
@@ -2021,7 +2021,13 @@ void Interpolate::initSupportedPrimitiveDescriptors() {
         return;
 
     ov::element::Type inputPrecision = getOriginalInputPrecisionAtPort(DATA_ID);
-    if ((inputPrecision != ov::element::i8) && (inputPrecision != ov::element::u8) && (inputPrecision != ov::element::bf16)) {
+
+#if defined(OV_CPU_WITH_ACL)
+    bool isInputPrecisionSupported = one_of(inputPrecision, ov::element::i8, ov::element::u8, ov::element::bf16, ov::element::f16);
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:2026]: I don't see any limitations on supported precision inside ACL Interpolate Executor (btw why?), but I believe bf16 shouldn't be in the list. Maybe smt else.
BTW does Intepolate test on ARM covers all these precisions?

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp [hunk 5]
```diff
@@ -2039,9 +2045,11 @@ void Interpolate::initSupportedPrimitiveDescriptors() {
         outputPrecision = fusedWith[fusedWith.size() - 1]->getOriginalOutputPrecisionAtPort(DATA_ID);
     }
 
+#if defined(OPENVINO_ARCH_X86_64)
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:2048]: I woul prefer to keep this condition for risc-v. So we should skip it only for OV_CPU_WITH_ACL case

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp (hunk 1 above)
Comment (alvoron) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp:2128]: Fixed

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp (hunk 2 above)
Comment (alvoron) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp:2124]: Fixed

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp (hunk 3 above)
Comment (alvoron) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp:2129]: Added

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp (hunk 4 above)
Comment (alvoron) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp:2026]: Right, bf16 is removed, i16 is added.
List of supported ACL data type configuration here: https://github.com/ARM-software/ComputeLibrary/blob/main/arm_compute/runtime/NEON/functions/NEScale.h#L60-L69

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp (hunk 5 above)
Comment (alvoron) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp:2048]: Fixed

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp (hunk 4 above)
Comment (alvoron) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: SLT interpolate tests are not enabled on ARM.

**Comment by alvoron** [comment]
f16 Interpolate is temporary disabled on ARM because of ACL f16 NEScale performance issue: https://github.com/ARM-software/ComputeLibrary/issues/1134
Performance degradation has been observed on `fast-neural-style-mosaic`.

**Comment by alvoron** [comment]
The PR should not be merged until https://github.com/ARM-software/ComputeLibrary/issues/1134 is fixed

**Comment by alvoron** [comment]
ACL issue mentioned above happens with `NCHW` only. ACL preferred layout is `NHWC` and this layout is prioritised in Interpolate node for ARM platforms.
I retested `fast-neural-style-mosaic` and `fastseg-small`, in both models ACL f16 Interpolate latency lower than f32 Interpolate latency.
@dmitry-gorokhov  I assume, we may merge this PR.

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**